include/z64save.h

```diff
@@ -4,6 +4,28 @@
 #include "ultra64.h"
 #include "z64math.h"
 
+typedef enum {
+    /*  0 */ HUD_VISIBILITY_IDLE,
+    /*  1 */ HUD_VISIBILITY_NONE,
+    /*  2 */ HUD_VISIBILITY_NONE_ALT, // Identical to HUD_VISIBILITY_NONE
```

Comment:
even though it makes sense in hindsight, I had to think a little bit about what _NONE on its own meant for visibility.
wondering if for this (and for all of them, really) if SHOW could help make it more fluid. `_SHOW_NONE`, `_SHOW_ALL`, `_SHOW_HEARTS_MAGIC`, etc

on the contrary `_NONE` can also be `_HIDE_ALL`, which would probably have made it even easier for me to understand, but I can understand that seeming a bit backwards to some

dunno, just brainstorming ideas, I think these names are good and would work fine. but since this system sucks I think the least we can do is make it super clear and easy to understand from a user-facing external perspective

Reply:
using SHOW (and shortening visibility so the enum is ginormous):
![image](https://user-images.githubusercontent.com/22045047/202360793-136734c5-6b11-4c13-8232-05ebacc308f9.png)

and using HIDE for NONE (currently my favorite i think) 
![image](https://user-images.githubusercontent.com/22045047/202360856-55c824e7-0e24-4415-8cd6-82a3409c3b51.png)

Reply:
Not a fan tbh, and I say that despite curbing my opinions to not delay this pr further. I can agree for the sake of merging though

Imo "visibility" is enough for being related to show/hide (or at least the way you have it may as well go back to "mode")

hide_all / show_none which is the only use for "hide" doesn't sound worth making a distinction.
Would "nothing" instead of "none" help idk? (visibility_nothing)

Reply:
I guess I do see it as a mode or a setting. Specifically a visibility mode though. My issue with hud mode in the past was that "hud mode" on its own sounded too broad, relating to the state of the hud as a whole, not just the alpha/visibility. "setting a visibility" doesnt make as much sense to me as "setting a visibility mode/setting", which involves showing or hiding certain things

Im not here to push changes through/"settle for the sake of merging". I get the PR has taken a while thus far, but I just want to get my thoughts out there for what makes sense to me, especially since i've had time to let my thoughts brew on the subject. This is a pretty important change thats being made, and dont think the discussion on it should feel rushed

Reply:
Sorry, I don't mean discussion shouldn't happen, just that I'm personally kind of out of it, like I personally don't really care as long as this gets merged

Feel free to discuss changes of course

Reply:
I'm fine with these discussions, I'm just interested in finding the best names.

`SHOW` is interesting to me, I wonder if it could even drop the `_VISI_` and those macros still make sense. But then if there's both `_SHOW_` and `_HIDE_`, you'd still need that common name between them for the function `Interface_ChangeHudVisibility` and give it a way to tie those macros to that function.

Reply:
yeah using just show is interesting. though if we're looking at is as a normal "system" prefix, just "hud show" doesnt really make sense.
But, if you say that the system is still "hud visibility", but it just has enums for the different "modes" maybe its fine.

i also wouldnt mind if we were more aggressive about calling these "modes", ie `hudVisibilty` -> `hudVisibilityMode` and such. that may or may not help defines like `HUD_SHOW_X` make more sense (since `HUD_SHOW_` is just describing the mode). 
But like I said before, I would still be ok keeping everything as they are now. Just putting out other options that we may like more